easybuild/scripts/findPythonDeps.py

```diff
@@ -0,0 +1,148 @@
+#!/usr/bin/env python
+
+import argparse
+import json
+import os
+import re
+import shutil
+import subprocess
+import sys
+import tempfile
+from contextlib import contextmanager
+from pprint import pprint
+try:
+    import pkg_resources
+except ImportError as e:
+    print('pkg_resources could not be imported: %s\nYou might need to install setuptools!' % e)
+    sys.exit(1)
+
+
+@contextmanager
+def temporary_directory(*args, **kwargs):
+    """Resource wrapper over tempfile.mkdtemp"""
+    name = tempfile.mkdtemp(*args, **kwargs)
+    try:
+        yield name
+    finally:
+        shutil.rmtree(name)
+
+
+def extract_pkg_name(package_spec):
+    return re.split('<|>|=|~', args.package, 1)[0]
+
+
+def run_cmd(arguments, action_desc, **kwargs):
+    """Run the command and return the return code and output"""
+    extra_args = kwargs or {}
+    if sys.version_info[0] >= 3:
+        extra_args['universal_newlines'] = True
+    p = subprocess.Popen(arguments, stdout=subprocess.PIPE, stderr=subprocess.STDOUT, **extra_args)
+    out, _ = p.communicate()
+    if p.returncode != 0:
+        raise RuntimeError('Failed to %s: %s' % (action_desc, out))
+    return out
+
+
+def run_in_venv(cmd, venv_path, action_desc):
+    """Run the givven command in the virtualenv at the given path"""
+    cmd = 'source %s/bin/activate && %s' % (venv_path, cmd)
+    return run_cmd(cmd, action_desc, shell=True, executable='/bin/bash')
+
+
+def get_dep_tree(package_spec):
+    """Get the dep-tree for installing the given Python package spec"""
+    package_name = extract_pkg_name(package_spec)
+    with temporary_directory(suffix=package_name + '-deps') as tmp_dir:
+        # prevent pip from (ab)using $HOME/.cache/pip
+        os.environ['XDG_CACHE_HOME'] = os.path.join(tmp_dir, 'pip-cache')
+        venv_dir = os.path.join(tmp_dir, 'venv')
+        # create virtualenv, install package in it
+        run_cmd(['virtualenv', '--system-site-packages', venv_dir], action_desc='create virtualenv')
```

Comment:
There should be a check somewhere that `virtualenv` is available. I'm guessing there may also be assumptions here about the  Python version running this, this should probably also be verified

Reply:
There shouldn't be any particular assumptions here. But added checks for `eb` and `virtualenv`